front/src/common/Map/DefaultBaseMap.tsx

```diff
@@ -41,38 +41,36 @@ const DefaultBaseMap = ({
   pathStepMarkers = [],
   isFeasible = true,
   children,
-  layersSettings,
+  mapSettings,
+  updateMapSettings,
   highlightedArea,
 }: PropsWithChildren<DefaultBaseMapProps>) => {
   const mapRef = useRef<MapRef | null>(null);
-  const [viewPort, setViewPort] = useState(mapInitialState.viewport);
+
```

Comment:
Could you extract viewport from the mapSettings and then replace all the `mapSettings.viewport` by `viewport` ? This will reduce the number of changed lines in this files
```suggestion
const { viewport } = mapSettings;

```